src/proto/flwr/proto/recordset.proto

```diff
@@ -0,0 +1,36 @@
+// Copyright 2024 Flower Labs GmbH. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ==============================================================================
+
+syntax = "proto3";
+
+package flwr.proto;
+
+import "flwr/proto/task.proto";
+
+message Array {
+  string dtype = 1;
+  repeated int32 shape = 2;
+  string stype = 3;
+  bytes data = 4;
+}
+
+message ParametersRecord {
+  repeated string data_keys = 1;
+  repeated Array data_values = 2;
+}
+
+message MetricsRecord { map<string, Value> data = 1; }
```

Comment:
`MetricsRecord` only allows `int` and `float` types (and lists of those two types)

Reply:
Yes, I get it.
I was thinking that, since we enforce type checks when instantiating the `MetricsRecord`, we can reuse `Value` to store them to avoid having repetitive code in proto files and in `serde.py`. But I also happy with copy-pasting part of the serde functions for `Value` and create new messages for `MetricsRecordValue`.

Reply:
Got it, makes sense. I suspected that to be the case :)

My recommendation would be to create new messages that map closely to the Python code for two reasons:
1. Enable us to remove `Value` once we migrated everything to `RecordSet`
2. Make it easier to implement non-Python clients based on the ProtoBuf definitions